### DISCUSSION TITLE
Sensor values not updated in HA (closed, replaced by issue)

### DISCUSSION THREAD

**Discussion by DreamT80** [flobz/psa_car_controller#191] [Q&A]
(I made an issue on this instead, can be found [here](https://github.com/flobz/psa_car_controller/issues/199))

Hi,

I'm using PSA Car Controller with HA and my Opel Grandland.  I've successfully connected it to my accounts and added some sensors using [this](https://github.com/Flodu31/HomeAssistant-PeugeotIntegration) guide.

But, the sensor values are only updated if I start/restart psa-cc. It's like it only reads the values on startup. Though, the log hints that calls are being made regularly, so it seems like it works.

Does anyone know why? Is there a good way to debug this? It feels like it's cache related, but that's only a guess.

**Comment by DreamT80** [comment]
I looked into it a bit further and found this in the log. It happens during __refresh_vehicle_info, so I guess that's the reason for the values not getting updated:

`172.30.32.1 - - [23/Jul/2021 14:44:06] "GET /charge_control?vin=[VIN]&?always_check=true HTTP/1.1" 200 -
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/psa_car_controller/my_psacc.py", line 169, in __refresh_vehicle_info
    self.get_vehicle_info(car.vin)
  File "/psa_car_controller/my_psacc.py", line 156, in get_vehicle_info
    self.record_info(car)
  File "/psa_car_controller/my_psacc.py", line 460, in record_info
    mileage = car.status.timed_odometer.mileage
AttributeError: 'NoneType' object has no attribute 'mileage'`

Looking into the JSON object returned from the service shows that timed_odometer is null, thus haven't got mileage. Maybe that is the reason?

`"timed_odometer":null`

Maybe that value isn't available for my car model?

I'd like to disable that part of the code and test locally, but I don't know where to find my_psacc.py in my system...

**Comment by DreamT80** [comment]
@flobz, maybe the issue I'm talking about is fixed in this commit? 
https://github.com/flobz/psa_car_controller/commit/c4486f7ae75ae5a7429e3fdcae696a7ec89663aa

I'd like to try it out myself, but I really don't know how to reach the code. I know that it's running in a container, but I can't find it in the host file system...(running HassOS)

**Reply by DreamT80** [reply]
@flobz , can you please help me with this?

**Comment by DavidFFerreira** [comment]
@DreamT80 I have several issues with HA integration, and already understant that anything related with HA @flobz will not help

**Reply by DreamT80** [reply]
This issue is closed, and the issue is solved in another tread :).

But as you say, @flobz won’t help with HA related issues, which I understand since his work with this relates to the integration itself. Regarding this he is very helpful.
